plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/controller/CodeTransformChatController.kt

```diff
@@ -359,6 +387,164 @@ class CodeTransformChatController(
+            hilTryResumeAfterError(message("codemodernizer.chat.message.hil.error.no_other_versions_found", hilDownloadArtifact.manifest.pomArtifactId))
+            return
+        }
+
+        val dependency = codeModernizerManager.findAvailableVersionForDependency(
+            hilDownloadArtifact.manifest.pomGroupId,
+            hilDownloadArtifact.manifest.pomArtifactId
+        )
+
+        if (dependency == null || (dependency.majors.isNullOrEmpty() && dependency.minors.isNullOrEmpty() && dependency.incrementals.isNullOrEmpty())) {
+            hilTryResumeAfterError(message("codemodernizer.chat.message.hil.error.no_other_versions_found", hilDownloadArtifact.manifest.pomArtifactId))
+            return
+        }
+        codeTransformChatHelper.updateLastPendingMessage(buildTransformAwaitUserInputChatContent(dependency))
+        runInEdt {
+            codeModernizerManager.getBottomToolWindow().show()
+        }
+    }
+
+    // Remove open file button after pom.xml is deleted
+    private suspend fun updatePomPreviewItem() {
+        val hilPomItemId = codeTransformChatHelper.getHilPomItemId() ?: return
+        val hilDownloadArtifact = codeModernizerManager.getArtifactForHil()
+        if (hilDownloadArtifact != null) {
+            codeTransformChatHelper.updateExistingMessage(
+                hilPomItemId,
+                buildTransformDependencyErrorChatContent(hilDownloadArtifact, false)
+            )
+        }
+        codeTransformChatHelper.clearHilPomItemId()
+    }
+
+    override suspend fun processConfirmHilSelection(message: IncomingCodeTransformMessage.ConfirmHilSelection) {
+        if (!checkForAuth(message.tabId)) {
+            return
+        }
+
+        updatePomPreviewItem()
+
+        val selectedVersion = message.version
+        val artifact = codeModernizerManager.getCurrentHilArtifact() as CodeTransformHilDownloadArtifact
+
+        codeTransformChatHelper.run {
+            addNewMessage(buildUserHilSelection(artifact.manifest.pomArtifactId, artifact.manifest.sourcePomVersion, selectedVersion))
+
+            addNewMessage(buildCompileHilAlternativeVersionContent())
+        }
+
+        val copyDependencyResult = codeModernizerManager.copyDependencyForHil(selectedVersion)
+        if (copyDependencyResult == MavenCopyCommandsResult.Failure) {
+            hilTryResumeAfterError(message("codemodernizer.chat.message.hil.error.cannot_upload"))
+            return
+        } else if (copyDependencyResult == MavenCopyCommandsResult.Cancelled) {
+            hilTryResumeAfterError(message("codemodernizer.chat.message.hil.error.cancel_upload"))
+            return
+        }
+
+        try {
+            codeModernizerManager.tryResumeWithAlternativeVersion(selectedVersion)
+
+            telemetry.logHil(
+                CodeModernizerSessionState.getInstance(context.project).currentJobId?.id as String,
+                HilTelemetryMetaData(
+                    versionSelected = selectedVersion,
```

Comment:
Blocker: VSCode has `dependencyVersionSelected` as `versionSelected` might have overlap with Java version or other "versions" selected in the future.

Are we using `dependencyVersionSelected` or `versionSelected`?